guizero/Waffle.py

```diff
@@ -106,34 +168,22 @@ def get_all(self):
             row = []
             for x in range(self._width):
                 obj = self._save_canvas[y][x]
-                row.append(self._canvas.itemcget(obj,'fill'))
+                row.append(self._canvas.itemcget(obj, "fill"))
             all_pixels.append(row)
         return all_pixels
 
     # Internal use only
     # Detect x,y coords of where the user clicked
-    def _clicked_on(self,e):
+    def _clicked_on(self, e):
         # you can only click on the waffle if its enabled
         if self._enabled:
             canvas = e.widget
             x = canvas.canvasx(e.x)
             y = canvas.canvasy(e.y)
-            pixel_x = int(x/(self._pixel_size+self._pad))
-            pixel_y = int(y/(self._pixel_size+self._pad))
+            pixel_x = int(x / (self._pixel_size + self._pad))
+            pixel_y = int(y / (self._pixel_size + self._pad))
             if self._command:
-                args_expected = utils.no_args_expected(self._command)
-                if args_expected == 0:
-                    self._command()
-                elif args_expected == 2:
-                    self._command(pixel_x,pixel_y)
-                else:
-                    utils.error_format("Waffle command function must accept either 0 or 2 arguments.\nThe current command has {} arguments.".format(args_expected))
-
-    def update_command(self, command):
-        if command is None:
-            self._command = lambda: None
-        else:
-            self._command = command
+                self._command(pixel_x, pixel_y)
```

Comment:
you should be using ._update_command here which you seem to have removed? perhaps your branch was out of date?